ResearchSuite/ResearchSuite/RSDTaskInfoStepObject.swift

```diff
@@ -67,12 +50,89 @@ public struct RSDTaskInfoStepObject : RSDTaskInfoStep, RSDEmbeddedIconVendor, De
     public var detail: String?
     
     /// The estimated number of minutes that the task will take. If `0`, then this is ignored.
-    public var estimatedMinutes: Int = 0
+    public var estimatedMinutes: Int {
+        return _estimatedMinutes ?? 0
+    }
+    private var _estimatedMinutes: Int?
     
     /// The icon used to display this task reference in a list of tasks.
     public var icon: RSDImageWrapper?
     
+    private enum CodingKeys : String, CodingKey {
+        case identifier, title, subtitle, detail, _estimatedMinutes = "estimatedMinutes", icon
+    }
+    
+    public init(identifier: String) {
+        self.identifier = identifier
+    }
+    
+    public func copy(with identifier: String) -> RSDTaskInfoObject {
+        var copy = RSDTaskInfoObject(identifier: identifier)
+        copy.title = self.title
+        copy.subtitle = self.subtitle
+        copy.detail = self.detail
+        copy._estimatedMinutes = self._estimatedMinutes
+        copy.icon = self.icon
+        return copy
+    }
+}
+
+extension RSDTaskInfoObject : Equatable {
+    public static func ==(lhs: RSDTaskInfoObject, rhs: RSDTaskInfoObject) -> Bool {
+        return lhs.identifier == rhs.identifier &&
+            lhs.title == rhs.title &&
+            lhs.subtitle == rhs.subtitle &&
+            lhs.detail == rhs.detail &&
+            lhs.estimatedMinutes == rhs.estimatedMinutes &&
+            lhs.icon == rhs.icon
+    }
+}
+
+/// `RSDTaskInfoStepObject` is a concrete implementation of the `RSDTaskInfoStep` protocol.
+public struct RSDTaskInfoStepObject : RSDTaskInfoStep, Decodable {
+
+    private enum CodingKeys : String, CodingKey {
+        case stepType = "type", schemaIdentifier, schemaRevision, taskTransformer
+    }
+
+    /// Returns the task info identifier.
+    public var identifier: String {
+        return taskInfoObject.identifier
```

Comment:
For consistency, should this be self.taskInfoObject.identifier? (also on the imageVendor var, below)